### PR TITLE
Update intro.md

### PR DESCRIPTION
**Pull request by Atcold**
Refactored text for consistency with the rest of the doc.
The goal of training a nn is to perform well on the validation set, not the training set.
Removed `local` from snippet, so they are runnable in the interpreter.

### CODE REVIEW

**Comment by Atcold** [comment]
I'm still going though the doc. I'll post here when it's safe merging.

**Comment by Atcold** [comment]
Done, @soumith. Documentation fixed and updated. Feel free to merge.

**Comment by soumith** [comment]
thanks!
